tla-types/src/test/scala/at/forsyte/apalache/tla/typecheck/etc/TestTypeCheckerTool.scala

```diff
@@ -0,0 +1,69 @@
+package at.forsyte.apalache.tla.typecheck.etc
```

Comment:
@konnov Seems like we should also add tests for the many Tla modules that are part of this PR. 

Do you think it makes sense to do those as integration tests via our mdx setup? Or should I put them here?

Reply:
The PR has several `*.tla` files that can be used for integration testing. Yes, it would be good to have them integrated in the mdx setup.

Reply:
So currently type checking is failing on these files:

Reply:
Yes, it was definitely failing on `SlidingPuzzlesTyped` and `QueensTyped`. `QueensTyped` definitely require another operator that turns a function into a sequence. I don't remember about `TwoPhaseTyped.tla`; that is a bit surprising, but something was probably tricky there. Let's not fix these ones.

Reply:
I fixed the errors on `SlidingPuzzles` and `TwoPhased` and added a failing test for the `Queens` spec, which we can fix once support for the needed sub-typing or coercion is supplied.